prolif/interactions/interactions.py

```diff
@@ -559,3 +563,482 @@ def detect(
+        Tolerance for the deviation from the ideal acceptor atom's angle (degrees).
+        If the deviation is larger than this value, the interaction will not be
+        considered valid (geometry checks).
+    tolerance_dev_daa : float
+        Tolerance for the deviation from the ideal donor atom's angle (degrees).
+        If the deviation is larger than this value, the interaction will not be
+        considered valid (geometry checks).
+    tolerance_dev_apa : float
+        Tolerance for the deviation from the ideal acceptor plane angle (degrees).
+        If the deviation is larger than this value, the interaction will not be
+        considered valid (geometry checks).
+    tolerance_dev_dpa : float
+        Tolerance for the deviation from the ideal donor plane angle (degrees).
+        If the deviation is larger than this value, the interaction will not be
+        considered valid (geometry checks).
+    vdwradii : dict[str, float] | None
+        Custom van der Waals radii for elements, if not provided, the default preset
+        will be used (to calculate `vina_hbond_potential`).
+    vdwradii_preset : Literal["mdanalysis", "rdkit", "csd"]
+        Preset for van der Waals radii (to calculate `vina_hbond_potential`). Defaults
+        to "csd". The presets are defined in
+        :mod:`prolif.interactions.constants.VDW_PRESETS`.
+    vina_hbond_potential_g : float
+        H-bond potential to indicate the good distance of forming a hydrogen bond
+        (derived from AutoDock Vina's H-bond terms).
+    vina_hbond_potential_b : float
+        H-bond potential to indicate the bad distance of forming a hydrogen bond
+        (derived from AutoDock Vina's H-bond terms).
+    ignore_geometry_checks : bool
+        If True, the geometry checks for the interaction will be skipped. This is useful
+        for cases where the geometry is not relevant or when the user wants to skip the
+        geometry checks for performance reasons. Defaults to False.
+
+    """
+
+    def __init__(
+        self,
+        acceptor: str = (
+            "[$([N&!$([NX3]-*=[O,N,P,S])&!$([NX3]-[a])&!$([Nv4+1])&!$(N=C(-[C,N])-N)])"
+            ",$([n+0&!X3&!$([n&r5]:[n+&r5])])"
+            ",$([O&!$([OX2](C)C=O)&!$(O(~a)~a)&!$(O=N-*)&!$([O-]-N=O)])"
+            ",$([o+0])"
+            ",$([F&$(F-[#6])&!$(F-[#6][F,Cl,Br,I])])]"
+        ),
+        donor: str = (
+            # implicit or explicit hydrogen bond donors
+            "[$([O,S,#7;+0&h,+0&H,+0&H2,+0H3])"
+            ",$([N;v4&+1&h,v4&+1&H,v4&+1&H2,v4&+1&H3,v4&+1&H4])"
+            ",$([n+]c[nh]),$([n+]c[nH])]"
+        ),
+        distance: float = 3.5,
+        include_water: bool = False,
+        tolerance_dev_aaa: float = 90,
+        tolerance_dev_daa: float = 25,
+        tolerance_dev_apa: float = 90,
+        tolerance_dev_dpa: float = 30,
+        vdwradii: dict[str, float] | None = None,
+        vdwradii_preset: Literal["mdanalysis", "rdkit", "csd"] = "csd",
```

Comment:
I'd be tempted to just **not** expose these parameters, the CSD dict should be comprehensive enough so there's no reason to expose `vdwradii`, and the preset was used to tune the vina potential parameters so changing it is really not advised